### PR TITLE
Log Vulkan loader errors if the instance failed creation on Fuchsia

### PR DESCRIPTION
**Pull request by jbauman42**
Log Vulkan loader errors if the instance failed creation on Fuchsia
The vulkan loader can output logs to help us debug why instance creation
fails. To catch these logs, we need to pass a debug report callback to
vkCreateInstance (since the only other debug report callback is set up
after the instance is created).

Outputting logs is currently only enabled on Fuchsia, since other
platforms may have fallbacks and wouldn't want the error logspam.

Fixes https://github.com/flutter/flutter/issues/82928

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by goderbauer** [comment]
(triage) I spoke to @jbauman42 last week and they indicated plans to come back to this at some point.

File: vulkan/vulkan_application.cc [hunk 1]
```diff
@@ -96,10 +127,19 @@ VulkanApplication::VulkanApplication(
 
   if (VK_CALL_LOG_ERROR(vk_.CreateInstance(&create_info, nullptr, &instance)) !=
       VK_SUCCESS) {
+#if OS_FUCHSIA
```
Comment (chinmaygarde) [inline vulkan/vulkan_application.cc:130]: FYI: This library is only used on Fuchsia at the moment. I don't think there is a need to have Fuchsia specific guards.

Comment (jbauman42) [reply vulkan/vulkan_application.cc]: I didn't realize that. I've combined the log into the existing one and removed the #if.

**Review by chinmaygarde — APPROVED** [review]

**Comment by chinmaygarde** [comment]
Wilco. Fetching a second reviewer. Consider go/flutter-onboarding to become a contributor 🙏🏽

**Review by jtmcdole — APPROVED** [review]

**Pull request merged**